packages/react-components/react-color-picker-preview/library/src/components/ColorArea/useColorAreaStyles.styles.ts

```diff
@@ -1,22 +1,73 @@
-import { makeStyles, mergeClasses } from '@griffel/react';
+import { makeResetStyles, makeStyles, mergeClasses } from '@griffel/react';
 import type { SlotClassNames } from '@fluentui/react-utilities';
+import { tokens } from '@fluentui/react-theme';
 import type { ColorAreaSlots, ColorAreaState } from './ColorArea.types';
 
 export const colorAreaClassNames: SlotClassNames<ColorAreaSlots> = {
   root: 'fui-ColorArea',
-  // TODO: add class names for all slots on ColorAreaSlots.
-  // Should be of the form `<slotName>: 'fui-ColorArea__<slotName>`
+  thumb: 'fui-ColorArea__thumb',
 };
 
+export const colorAreaCSSVars = {
+  areaXProgressVar: `--fui-AreaX--progress`,
+  areaYProgressVar: `--fui-AreaY--progress`,
+  areaStepsPercentVar: `--fui-Area--steps-percent`,
+  thumbColorVar: `--fui-Area__thumb--color`,
+  mainColorVar: `--fui-Area--main-color`,
+};
+
+// Internal CSS variables
+const thumbSizeVar = `--fui-Slider__thumb--size`;
+
+const { areaXProgressVar, areaYProgressVar, thumbColorVar, mainColorVar } = colorAreaCSSVars;
+
 /**
  * Styles for the root slot
  */
-const useStyles = makeStyles({
-  root: {
-    // TODO Add default styles for the root element
-  },
+const useRootStyles = makeResetStyles({
+  position: 'relative',
+  border: `1px solid ${tokens.colorNeutralStroke1}`,
+  borderRadius: tokens.borderRadiusMedium,
+  background: `linear-gradient(to bottom, transparent, #000), linear-gradient(to right, white, transparent), var(${mainColorVar})`,
+  'forced-color-adjust': 'none',
+  display: 'inline-grid',
+  touchAction: 'none',
+  alignItems: 'start',
+  justifyItems: 'start',
+  [thumbSizeVar]: '20px',
+  minWidth: '200px',
+  minHeight: '200px',
+});
 
-  // TODO add additional classes for different states and/or slots
+/**
+ * Styles for the thumb slot
+ */
+const useThumbStyles = makeStyles({
+  thumb: {
+    position: 'absolute',
+    width: `var(${thumbSizeVar})`,
+    height: `var(${thumbSizeVar})`,
+    pointerEvents: 'none',
+    outlineStyle: 'none',
+    forcedColorAdjust: 'none',
+    borderRadius: tokens.borderRadiusCircular,
+    boxShadow: `0 0 0 calc(var(${thumbSizeVar}) * .2) ${tokens.colorNeutralBackground1} inset`,
+    backgroundColor: `var(${thumbColorVar})`,
+    transform: 'translate(-50%, 50%)',
```

Comment:
I'm curious, why is the offset for X negative while it's positive for Y? Wouldn't both axes need a -50% offset to center the element properly?

Reply:
It won't work with -50%
![color-area-2](https://github.com/user-attachments/assets/e277cbaf-086e-4308-a100-8bb077591637)

Reply:
> It won't work with -50% ![color-area-2](https://private-user-images.githubusercontent.com/11574680/374922940-e277cbaf-086e-4308-a100-8bb077591637.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2NDU2NDQsIm5iZiI6MTcyODY0NTM0NCwicGF0aCI6Ii8xMTU3NDY4MC8zNzQ5MjI5NDAtZTI3N2NiYWYtMDg2ZS00MzA4LWExMDAtOGJiMDc3NTkxNjM3LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDExVDExMTU0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBkN2JhMmY1MzQyMjQyMWQ4MDA3ZGJjMjI5YWZkMzk4ODU0ZDM5ZjRkNmY1M2FhMGFhNzhmNWRlOGMyOGRjZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9zGV_KCoJWX_vm8Gz2ejKoP-Y2_5fUwuW0MsMKUIL08) [ ![color-area-2](https://private-user-images.githubusercontent.com/11574680/374922940-e277cbaf-086e-4308-a100-8bb077591637.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2NDU2NDQsIm5iZiI6MTcyODY0NTM0NCwicGF0aCI6Ii8xMTU3NDY4MC8zNzQ5MjI5NDAtZTI3N2NiYWYtMDg2ZS00MzA4LWExMDAtOGJiMDc3NTkxNjM3LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDExVDExMTU0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBkN2JhMmY1MzQyMjQyMWQ4MDA3ZGJjMjI5YWZkMzk4ODU0ZDM5ZjRkNmY1M2FhMGFhNzhmNWRlOGMyOGRjZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9zGV_KCoJWX_vm8Gz2ejKoP-Y2_5fUwuW0MsMKUIL08) ](https://private-user-images.githubusercontent.com/11574680/374922940-e277cbaf-086e-4308-a100-8bb077591637.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2NDU2NDQsIm5iZiI6MTcyODY0NTM0NCwicGF0aCI6Ii8xMTU3NDY4MC8zNzQ5MjI5NDAtZTI3N2NiYWYtMDg2ZS00MzA4LWExMDAtOGJiMDc3NTkxNjM3LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDExVDExMTU0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBkN2JhMmY1MzQyMjQyMWQ4MDA3ZGJjMjI5YWZkMzk4ODU0ZDM5ZjRkNmY1M2FhMGFhNzhmNWRlOGMyOGRjZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9zGV_KCoJWX_vm8Gz2ejKoP-Y2_5fUwuW0MsMKUIL08) [ ](https://private-user-images.githubusercontent.com/11574680/374922940-e277cbaf-086e-4308-a100-8bb077591637.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2NDU2NDQsIm5iZiI6MTcyODY0NTM0NCwicGF0aCI6Ii8xMTU3NDY4MC8zNzQ5MjI5NDAtZTI3N2NiYWYtMDg2ZS00MzA4LWExMDAtOGJiMDc3NTkxNjM3LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDExVDExMTU0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBkN2JhMmY1MzQyMjQyMWQ4MDA3ZGJjMjI5YWZkMzk4ODU0ZDM5ZjRkNmY1M2FhMGFhNzhmNWRlOGMyOGRjZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9zGV_KCoJWX_vm8Gz2ejKoP-Y2_5fUwuW0MsMKUIL08)

I see, but I'm interested to know why there's a negative offset applied to only one axis, but not to another one.